src/Gateway/Voucher/MaybeDisableGateway.php

```diff
@@ -35,7 +35,7 @@ public function maybeDisableMealVoucherGateway(array $gateways)
         if (
         ($isWcApiRequest
             || !doing_action('woocommerce_payment_gateways')
-            || !wp_doing_ajax()
+            || (!wp_doing_ajax() && ! is_wc_endpoint_url('order-pay'))
```

Comment:
These conditions are difficult to understand.

Why we check `order-pay` only if not ajax? I mean I guess this page adds the gateways without ajax indeed, but before this change the `if` would trigger for all ajax requests, is it ok to remove this behavior?

Also the comment above probably needs to be updated.